bubble_sort.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+def bubble_sort_by(arr)
+  loop do
+    swap_count = 0
+
+    (0...arr.length).each do |i|
+      next_num = i + 1
+      next unless yield(arr[i], arr[next_num]) == 1
+
+      temp = arr[i]
```

Comment:
* [ ] bubble_sort_by sorts an array by accepting a block
The example below is failing.

  ```ruby
    unsorted = %w[hi hello hey]
    bubble_sort_by(unsorted) { |left, right| left.length - right.length } == %w[hi hey hello] # returns this error
     
    # NoMethodError:
    #   undefined method `length' for nil:NilClass
     # ./bubble_sort.rb:9:in `block (2 levels) in bubble_sort_by'
     # ./bubble_sort.rb:7:in `each'
     # ./bubble_sort.rb:7:in `block in bubble_sort_by'
     # ./bubble_sort.rb:4:in `loop'
     # ./bubble_sort.rb:4:in `bubble_sort_by'

  ```

- **[SUGGESTION]** the source of the error is probably because of the `next_num` counter, since when "i" reaches the last array item you will pass the item of index "next_num" to yield which is a nil item. :interrobang: